x-pack/plugins/fleet/server/services/epm/packages/install.ts

```diff
@@ -468,8 +469,31 @@ export async function installPackage(args: InstallPackageParams) {
   const logger = appContextService.getLogger();
   const { savedObjectsClient, esClient } = args;
 
+  const bundledPackages = await getBundledPackages();
+
   if (args.installSource === 'registry') {
     const { pkgkey, force, spaceId } = args;
+
+    const matchingBundledPackage = bundledPackages.find(
```

Comment:
@joshdover - I made this change to push the "fall back to bundled packages" logic down closer to the place we do the actual installation. The challenging part is the abstraction level of the actual `installPackage` function - it accept two signatures: one for registry packages and another for uploads. So, the choice I made here was to simply _always_ try to install a given package from its bundled equivalent if it exists. 

e.g. if `fleet_server-1.0.1` is provided, and we have that exact version included as a bundled package, we'll install from that source even if the function was called with `installSource: registry` in its arguments. I think this is a fine implementation choice and the only minor drawback I can think of would be snapshot packages where we have a bundled version downloaded to disk, but the version in the snapshot registry has been updated in-place. I think that's a minor enough edge case to warrant this choice. 

Let me know if I can clarify anything else here - I'd like to get your thoughts on this approach before merging. Thanks!

Reply:
Awesome, thanks for exploring this. I like this implementation much more. I'm ok with this drawback as well. One other minor drawback is that API requests to install a package at `latest` will not work, but I don't think we support this today anyways so it should be considered separately.

Could we add a test to `install.test.ts` for this case?